client/http-client/src/client.rs

```diff
@@ -207,16 +207,19 @@ impl ClientT for HttpClient {
 				}
 			};
 
-			let response: Response<_> = match serde_json::from_slice(&body) {
-				Ok(response) => response,
-				Err(_) => {
-					let err: ErrorResponse = serde_json::from_slice(&body).map_err(Error::ParseError)?;
-					return Err(Error::Call(CallError::Custom(err.error_object().clone().into_owned())));
-				}
-			};
+			let response: Response<serde_json::Value> =
+				match serde_json::from_slice::<Response<serde_json::Value>>(&body) {
+					Ok(response) => response,
+					Err(_) => {
+						let err: ErrorResponse = serde_json::from_slice(&body).map_err(Error::ParseError)?;
+						return Err(Error::Call(CallError::Custom(err.error_object().clone().into_owned())));
+					}
+				};
+
+			let result = serde_json::from_value(response.result).map_err(Error::ParseError)?;
```

Comment:
So, we deseriazlize the bytes into `Response<Value>` and then that `Value` into R?

It might be worth a comment explaining this split (where previously it deserialized all at once); somebody like me would look at this and think "ooh I can skip that Value step" and change it back!

Also, going via `Value` feels like quite a performance hit over going straight to `R`. If we need to do this in two steps, perhaps instead of going to `Value` we could go to `RawValue` to avoid the allocations, and then turn that into an `R`?